crates/ruff_python_formatter/src/statement/suite.rs

```diff
@@ -490,6 +503,40 @@ fn stub_file_empty_lines(
     }
 }
 
+/// Checks if an empty line should be inserted between the preceding and, optionally,
+/// the following node according to the [`blank_line_after_nested_stub_class`](https://github.com/astral-sh/ruff/issues/8891)
+/// preview style.
+///
+/// If `following` is `None`, then the preceding node is the last node in the suite.
+pub(crate) fn blank_line_after_nested_stub_class_condition(
+    preceding: AnyNodeRef<'_>,
+    following: Option<AnyNodeRef<'_>>,
+    f: &PyFormatter,
+) -> bool {
```

Comment:
This is the core condition which checks whether we need to add an empty line between the preceding and following node or not.